### PR TITLE
Internal: Implement minimize and expand of checklist popup [ED-15475]

### PR DESCRIPTION
**Pull request by ronenboxer**

### CODE REVIEW

**Review by davseve — APPROVED** [review]

File: tests/playwright/sanity/modules/checklist/checklist.test.ts [hunk 1]
```diff
@@ -285,4 +285,52 @@ test.describe( 'Launchpad checklist tests', () => {
...
+
+	test( 'Make sure checklist is expanded after previous test if it failed', async ( { page, apiRequests }, testInfo ) => {
+		const wpAdmin = new WpAdminPage( page, testInfo, apiRequests );
+
+		await wpAdmin.openNewPage();
+
+		const checklistHelper = new ChecklistHelper( page, testInfo, apiRequests );
+
+		await checklistHelper.toggleExpandChecklist( 'editor', true );
+	} );
```
Comment (elchugreeva) [inline tests/playwright/sanity/modules/checklist/checklist.test.ts:335]: why don't use mock data, so we don't toggle the expand list value in DB?

File: tests/playwright/sanity/modules/checklist/helper.ts [hunk 2]
```diff
@@ -22,6 +22,7 @@ export default class ChecklistHelper {
 		await this.editor.openUserPreferencesPanel();
 		await this.editor.setSwitcherControlValue( controlIds.preferencePanel.checklistSwitcher, shouldShow );
 		await this.page.waitForResponse( ( response ) => response.url().includes( 'wp-admin/admin-ajax.php' ), { timeout: 30000 } );
+		await this.page.waitForResponse( ( response ) => response.url().includes( 'wp-admin/admin-ajax.php' ), { timeout: 30000 } );
```
[the remark this answers is not in the document]
Comment (ronenelementor) [reply tests/playwright/sanity/modules/checklist/helper.ts:25]: I think so, one for the update and another one for the 'heartbeat' I guess, without it it wouldn't save the preferences and would become flaky

File: tests/playwright/sanity/modules/checklist/checklist.test.ts (hunk 1 above)
Comment (ronenelementor) [reply tests/playwright/sanity/modules/checklist/checklist.test.ts:335]: I want to check that the data is saved correctly in the DB and functions as expected in the front end

Comment (elchugreeva) [reply tests/playwright/sanity/modules/checklist/checklist.test.ts:335]: ok

File: tests/playwright/sanity/modules/checklist/helper.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (elchugreeva) [reply tests/playwright/sanity/modules/checklist/helper.ts:25]: ok

**Review by elchugreeva — APPROVED** [review]

**Pull request merged**